src/components/input/bl-input.ts

```diff
@@ -33,6 +33,9 @@ export default class BlInput extends FormControlMixin(LitElement) {
   @query('input')
   validationTarget: HTMLInputElement;
 
+  @query('slot[name="icon"]')
+  iconSlot: HTMLSlotElement;
```

Comment:
This can be a private getter instead of public property, like we did in Button component.

```ts
  private get _hasIconSlot() {
    return this.querySelector(':scope > [slot="icon"]') !== null;
  }
```